pkg/api/testapi/testapi.go

```diff
@@ -76,14 +77,21 @@ func SelfLink(resource, name string) string {
 	return fmt.Sprintf("/api/%s/%s/%s", Version(), resource, name)
 }
 
-// Returns the appropriate path for the given resource, namespace and name.
+// Returns the appropriate path for the given prefix (watch, proxy, redirect, etc), resource, namespace and name.
 // For ex, this is of the form:
-// /api/v1beta1/pods/pod0 for v1beta1 and
-// /api/v1beta3/namespaces/foo/pods/pod0 for v1beta3.
-func ResourcePath(resource, namespace, name string) string {
+// /api/v1beta1/watch/pods/pod0 for v1beta1 and
+// /api/v1beta3/watch/namespaces/foo/pods/pod0 for v1beta3.
+func ResourcePathWithPrefix(proxy, resource, namespace, name string) string {
```

Comment:
Might also consider shortening these names-- they make those table driven tests pretty hard to read :(